### PR TITLE
Adds OperatingSystem enum.

### PR DESCRIPTION
**Pull request by coollog**
This is useful for resolving which binary of `skaffold` to download.

### CODE REVIEW

File: skaffold-plugins-core/src/main/java/com/google/cloud/tools/skaffold/filesystem/OperatingSystem.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  /**
+   * Resolves the operating system type. *
+   *
+   * @return the {@link OperatingSystem}
+   */
+  public static OperatingSystem resolve() {
+    return resolve(System.getProperties());
+  }
+
+  static OperatingSystem resolve(Properties properties) {
```
Comment (chanseokoh) [inline skaffold-plugins-core/src/main/java/com/google/cloud/tools/skaffold/filesystem/OperatingSystem.java]: `@VisibleForTesting`

File: skaffold-plugins-core/src/main/java/com/google/cloud/tools/skaffold/filesystem/OperatingSystem.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.Properties;
+
+/** Gets information about the operating system. */
+public enum OperatingSystem {
+  LINUX,
+  MAC_OS,
+  WINDOWS;
+
+  /**
+   * Resolves the operating system type. *
```
Comment (chanseokoh) [inline skaffold-plugins-core/src/main/java/com/google/cloud/tools/skaffold/filesystem/OperatingSystem.java]: stranded `*` at the end

File: skaffold-plugins-core/src/test/java/com/google/cloud/tools/skaffold/filesystem/OperatingSystemTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  public void testLinux() {
+    Properties fakeProperties = new Properties();
+    fakeProperties.setProperty("os.name", "os is LiNuX");
+    Assert.assertEquals(OperatingSystem.LINUX, OperatingSystem.resolve(fakeProperties));
+  }
+
+  @Test
+  public void testMacOs() {
+    Properties fakeProperties = new Properties();
+    fakeProperties.setProperty("os.name", "os is mAc or DaRwIn");
```
Comment (chanseokoh) [inline skaffold-plugins-core/src/test/java/com/google/cloud/tools/skaffold/filesystem/OperatingSystemTest.java]: It'd be better to split this into two cases: mAc and DaRwIn.

File: skaffold-plugins-core/src/main/java/com/google/cloud/tools/skaffold/filesystem/OperatingSystem.java (hunk 1 above)
Comment (coollog) [reply skaffold-plugins-core/src/main/java/com/google/cloud/tools/skaffold/filesystem/OperatingSystem.java]: This is actually used by `UserCacheHome` (for testability). The other `resolve` will be useful for other classes that do not need fake out the `Properties`.

**Pull request merged**